internal/controllers/machinedeployment/mdutil/util.go

```diff
@@ -469,58 +475,71 @@ func FindNewMachineSet(deployment *clusterv1.MachineDeployment, msList []*cluste
 	// We deterministically choose the oldest new MachineSet with matching template hash.
```

Comment:
It's fine that the additional note on the func godoc was dropped, but maybe let's update this godoc here
(regarding "lets choose the MS with the most replicas so that there is minimum machine churn.")

I would also consider to stop linking that issue. No idea if that still makes sense